csharp/src/Drivers/Databricks/DatabricksStatement.cs

```diff
@@ -48,6 +48,20 @@ protected override void SetStatementProperties(TExecuteStatementReq statement)
             statement.CanDownloadResult = useCloudFetch;
             statement.CanDecompressLZ4Result = canDecompressLz4;
             statement.MaxBytesPerFile = maxBytesPerFile;
+
+            if (Connection.AreResultsAvailableDirectly())
+            {
+                statement.GetDirectResults = DatabricksConnection.defaultGetDirectResults;
+            }
+        }
+
+        /// <summary>
+        /// Checks if direct results are available.
+        /// </summary>
+        /// <returns>True if direct results are available and contain result data, false otherwise.</returns>
+        public bool HasDirectResults()
```

Comment:
```suggestion
        public bool HasDirectResults => DirectResults?.ResultSet != null && DirectResults?.ResultSetMetadata != null;
```
(This is a good candidate to be a property instead of a method.)